src/system_tray.rs

```diff
@@ -46,10 +47,23 @@ pub struct SystemTrayBuilder {
 }
 
 impl SystemTrayBuilder {
-  /// Creates a new SystemTray for platforms where this is appropriate.
+  /// Creates a new SystemTray with an empty identifier for platforms where this is appropriate.
   pub fn new(icon: Icon, tray_menu: Option<ContextMenu>) -> Self {
     Self {
       platform_tray_builder: SystemTrayBuilderPlatform::new(
+        TrayId::EMPTY,
+        icon,
+        tray_menu.map(|m| m.0.menu_platform),
+      ),
+      tooltip: None,
+    }
+  }
+
+  /// Creates a new SystemTray with an identifier for platforms where this is appropriate.
+  pub fn new_with_id(id: TrayId, icon: Icon, tray_menu: Option<ContextMenu>) -> Self {
```

Comment:
why not `with_id(self, id: TrayId) -> Self`

Reply:
You're right, good idea :D